assets/src/apps/delivery/components/ActivityRenderer.tsx

```diff
@@ -568,7 +568,7 @@ const ActivityRenderer: React.FC<ActivityRendererProps> = ({
     }
     const currentStateSnapshot: any = {};
     const appChanges = changes.changed.filter((change: any) => change.includes('app.'));
-    if (appChanges.length) {
+    if (appChanges.length && changes?.changed?.length > 1) {
```

Comment:
we seem to be going back and forth on this line. maybe we can create a unit test for this and/or document in comments why we are doing it so that we land on one way or another